scripts/register_saft_addresses.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Register SAFT accounts script
+ *
+ * ts-node scripts/register_saft_addresses.ts "network" "path to csv with addresses"
+ */
+
+import fs from 'fs'
+import { utils, constants, providers, Wallet } from 'ethers'
+import { TimeLockRegistry } from '../build/types/TimeLockRegistry'
+import { TimeLockRegistryFactory } from '../build/types/TimeLockRegistryFactory'
+import { TrustToken } from '../build/types/TrustToken'
+import { TrustTokenFactory } from '../build/types/TrustTokenFactory'
+
+export const txnArgs = { gasLimit: 2_000_000, gasPrice: 20_000_000_000 }
+
+export interface SaftAccount {
+  address: string,
+  amount: string,
+}
+
+const toTrustToken = (amount: string) => utils.parseUnits(amount, 8)
+const sum = (numbers: utils.BigNumber[]) => numbers.reduce((cumSum, value) => cumSum.add(value), constants.Zero)
+
+export const registerSaftAccounts = async (registry: TimeLockRegistry, trustToken: TrustToken, accounts: SaftAccount[]) => {
+  const totalAllowance = sum(accounts.map(({ amount }) => toTrustToken(amount)))
+  const tx = await trustToken.approve(registry.address, totalAllowance, txnArgs)
+  await tx.wait()
+  console.log('Transfers approved')
+  let { nonce } = tx
+  const pendingTransactions = []
+  for (const { address, amount } of accounts) {
+    pendingTransactions.push((await registry.register(address, toTrustToken(amount), { ...txnArgs, nonce: nonce + 1 })).wait()
```

Comment:
```suggestion
    pendingTransactions.push((await registry.register(address, toTrustToken(amount), { ...txnArgs, nonce: nonce++ })).wait()
```
and remove line 36?

Reply:
idk, i think your version is a bit harder to read

Reply:
Maybe... 😛